cmsis-core/cmsis_nvic.h

```diff
@@ -34,6 +34,12 @@
 
 #include "cmsis.h"
 
+/* throw meaningful compiler error if yotta configuration is not provided */
+#if !defined(YOTTA_CFG_CMSIS_NVIC_USER_IRQ_OFFSET)      || !defined(YOTTA_CFG_CMSIS_NVIC_USER_IRQ_NUMBER) || \
```

Comment:
@bogdanm is this symbol already correctly defined (defined for M3/4, !defined for M0/0+) or is it something to add? We could use the same conditions as [here](https://github.com/ARMmbed/cmsis-core/blob/master/source/cmsis_nvic.c#L38); although not beautiful it would be consistent (it is the same module)